### ISSUE TITLE
document PR git cleanup

### ISSUE THREAD

**Issue by mr-c** [dib-lab/khmer#1013]
Here is a faster method tested by @ctb and me. 

If there are merge commits then this approach is always easier than rebasing.

```
git pull # make sure the local is up to date
git pull dib master # get up to date
# fix any merge conflicts
git status # sanity check
git diff dib/master # does the diff look correct? (no merge markers)
git reset --soft dib/master # un-commit the differences from dib/master
git status # sanity check
git commit --all # package all differences in one commit
git status # sanity check
git push # should fail
git push --force # override what's in GitHub's copy of the branch/pull request
# Code review and/or merging from the pull request
```

(`git reset --soft` keeps the tracking for all newly added files )

**Comment by bocajnotnef** [comment]
+1 for having this formally documented

**Comment by mr-c** [comment]
Fixed in #1222
